### DISCUSSION TITLE
is there a way to get earnings and EPS estimate?

### DISCUSSION THREAD

**Discussion by fadai88** [ranaroussi/yfinance#2159] [General]
I'm getting YFNotImplementedError whenI use ".earnings" or "get_earnings()"?
Is there  a way to get historical earnings and their estimates?

**Comment by welcra** [comment]
```
import yfinance as yf

ticker = "AAPL"

print(yf.Ticker(ticker).earnings_history)
```
yfinance only gives earnings data for the last 4 quarters, use something else like finnhub or alpha vantage for older data
